grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryInfer.java

```diff
@@ -44,8 +44,8 @@
                 () -> new IllegalStateException(ErrorMessage.NO_GRAPH.getMessage())
         );
 
-        // TODO: Handle ordering
-        return new Reasoner(graph).resolveToQuery(inner.withGraph(graph)).stream();
+        // TODO: Don't materialise ALL the time
+        return new Reasoner(graph).resolveToQuery(inner.withGraph(graph), true).stream();
```

Comment:
@kasper-piskorski we should check with Haikal and Boris if we should materialise by default or not